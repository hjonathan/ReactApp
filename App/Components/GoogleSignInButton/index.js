import React, {Component} from "react";
import {
    Text,
    TouchableOpacity,
    View,
    Image
} from "react-native";
import Styles from "./styles";
import * as Res from "../../Assets/resources";

/**
 * Class GoogleSignInButton
 * Example:
 *
 * import GoogleSignInButton from '../GoogleSignInButton';
 * render() {
 *      return (
 *          <GoogleSignInButton
 *              onPress={() => {}}
 *              title="Title test"
 *          />
 *      );
 * }
 */
export default class Button extends Component {
    /**
     * Constructor GoogleSignInButton
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            title: (props && props.title) || "test",
            icon: Res.images.googleSignInIcon,
            params: (props && props.params) || {
                color: "red"
            }
        };
    }

    /**
     * Returns custom button adding the next elements:
     * 1. Google standard icons acoording to it's styleguide.
     * 2. Google standard label to sign in.
     * @param params
     * @returns {*}
     */
    renderCustomButton (params) {
        return (
            <View style={Styles.container}>
                <View style={Styles.customButton}>
                    <Image
                        source={this.state.icon}
                        style={Styles.icon}
                    />
                    <Text style={[Styles.label, {color: this.state.params.color}]}>
                        {this.state.title}
                    </Text>
                </View>
            </View>
        );
    }

    /**
     * Returns default button
     * @returns {*}
     */
    renderDefaultButton () {
        return (
            <View style={Styles.button}>
                <Text style={Styles.text}>
                    {this.state.title}
                </Text>
            </View>
        );
    }

    /**
     * Renders Component
     * @param onPress
     * @returns {*}
     */
    render ({onPress} = this.props) {
        return (
            <TouchableOpacity onPress={onPress}>
                {
                    (Object.prototype.hasOwnProperty.call(this.props, "params") && this.props.params)
                        ? this.renderCustomButton(this.props.params) : this.renderDefaultButton()
                }
            </TouchableOpacity>
        );
    }
}
