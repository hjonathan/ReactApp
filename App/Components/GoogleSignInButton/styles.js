import {StyleSheet} from "react-native";

export default StyleSheet.create({
    button: {
        alignItems: "center",
        borderRadius: 50,
        borderWidth: 2,
        borderColor: "#FFFFFF",
        paddingHorizontal: 50,
        paddingVertical: 10
    },
    text: {
        color: "#FFFFFF",
        fontWeight: "bold"
    },
    container: {
        paddingHorizontal: 14
    },
    icon: {
        flexBasis: 41,
        flexGrow: 0,
        flexShrink: 0
    },
    customButton: {
        backgroundColor: "#4285F4",
        flexDirection: "row",
        alignItems: "center",
        height: 41
    },
    label: {
        flex: 1,
        fontWeight: "bold",
        textAlign: "center",
        fontSize: 14
    }
});
