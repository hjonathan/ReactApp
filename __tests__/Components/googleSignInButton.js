import 'react-native';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import GoogleSignInButton from '../../App/Components/GoogleSignInButton';

describe('Component >>> GoogleSignInButton', () => {
    let defaultFlatButton,
        customFlatButton;

    const props = {
        params: {
            backgroundColor: "#4285F4",
            color: "#f0fc"
        },
        onPress: () => {}
    };

    beforeEach(() => {
        defaultFlatButton = shallow(<GoogleSignInButton params={props.params} onPress={props.onPress} />);
        customFlatButton = shallow(<GoogleSignInButton title={""} onPress={props.onPress} />);
    });

    it('Render DefaultFlatButton', () => {
        expect(toJson(defaultFlatButton)).toMatchSnapshot();
    });

    it('Render CustomFlatButton', () => {
        expect(toJson(customFlatButton)).toMatchSnapshot();
    });
});
