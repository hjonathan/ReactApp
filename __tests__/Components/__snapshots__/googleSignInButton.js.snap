// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component >>> GoogleSignInButton Render CustomFlatButton 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 50,
        "borderWidth": 2,
        "paddingHorizontal": 50,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontWeight": "bold",
        }
      }
    >
      test
    </Text>
  </Component>
</TouchableOpacity>
`;

exports[`Component >>> GoogleSignInButton Render DefaultFlatButton 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "paddingHorizontal": 14,
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4285F4",
          "flexDirection": "row",
          "height": 41,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../App/Assets/icons/googleSignIn/btn_google_dark_normal.png",
          }
        }
        style={
          Object {
            "flexBasis": 41,
            "flexGrow": 0,
            "flexShrink": 0,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "flex": 1,
              "fontSize": 14,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            Object {
              "color": "#f0fc",
            },
          ]
        }
      >
        test
      </Text>
    </Component>
  </Component>
</TouchableOpacity>
`;
